platform/linux-generic/odp_packet.c

```diff
@@ -699,6 +699,9 @@ void odp_packet_free(odp_packet_t pkt)
 
 static inline void packet_free_multi_ev(const odp_packet_t pkt[], int num, _odp_ev_id_t id)
 {
+	if (!num)
```

Comment:
This should use odp_unlikely(). Or maybe you could just have pkt_hdrs[num + 1] and avoid the branch altogether?

ODP API seems to allow odp_packet_free_multi() for zero packets, but I wonder if the fact that you bumped into such a free is a symptom of a bug in the calling code.

Reply:
The backtrace is
```
    #0 0x560f7aa63698 in packet_free_multi_ev /odp/platform/linux-generic/odp_packet.c:702
    #1 0x560f7aa63932 in odp_packet_free_multi /odp/platform/linux-generic/odp_packet.c:732
    #2 0x560f7aa0974d in free_pending_packets /odp/test/performance/odp_dmafwd.c:1128
    #3 0x560f7aa0a148 in process_packets /odp/test/performance/odp_dmafwd.c:1201
    #4 0x560f7aa0c86a in run_thread /odp/helper/threads.c:69
```
Which looks legitimate. When odp_dmafwd is exiting, it frees pending packets. If there aren't any for a given interface, it calls odp_packet_free_multi() with num=0, which is fine.